app/services/script-tags.ts

```diff
@@ -0,0 +1,118 @@
+
+export type Content = object | string | String | number | null | undefined;
+
+export type DataContent = Content | Content[];
+
+export interface ScriptTagsData {
+    src?: DataContent;
+    type?: DataContent;
+    content?: DataContent;
+}
+
+export interface ScriptTagDef {
+    [s: string]: DataContent;
+}
+
+export interface JSONLDScriptTagAttrs {
+    type: Content;
+    src: Content;
+}
+
+export type ScriptTagAttrs = JSONLDScriptTagAttrs;
+
+// ember-cli-meta-tags element types
+export enum TagType {
+    BASE = 'base',
+    LINK = 'link',
+    META = 'meta',
+    SCRIPT = 'script',
+    NOSCRIPT = 'noscript',
+    TITLE = 'title',
+}
+
+export interface HeadTagDef {
+    type: string;
+    content: DataContent;
+    attrs: MetaTagAttrs | ScriptTagAttrs;
+}
+
+/**
+ *
+ * Creates one script tag with model-specific overrides or default attribute values.
+ * Should additional script tags be needed, getHeadTags() should be called for each
+ * script and added to the headTags array.
+ *
+ * @method getScriptTagAttributes
+ * @method getHeadTags
+ * @param {ScriptTagsData} scriptTagsOverrides Overrides defaults
+ * @return {HeadTagsDefs} getHeadTags() returns HeadTagDefs for ScriptTagAttrs
+ */
+export default class ScriptTags extends Service {
+    @service intl!: Intl;
+    @service router!: any;
+    @service headTags!: HeadTagsService;
+
+    getScriptTagAttributes(scriptTagsOverrides: ScriptTagsData): ScriptTagDef {
+        // Default values
+        const scriptTagsData: ScriptTagsData = {
+            type: scriptTagsOverrides.type ?
+                scriptTagsOverrides.type : 'application/ld+json',
+            src: scriptTagsOverrides.src ?
+                scriptTagsOverrides.src : `osf.io/${identifier}/metadata/?format=google-dataset-json-ld`,
+            content: scriptTagsOverrides.content ?
+                scriptTagsOverrides.content : { isAccessibleForFree: true },
+            ...scriptTagsOverrides,
+        };
+        return {
+            type: scriptTagsData.type,
+            src: scriptTagsData.src,
+            dataContent: scriptTagsData.content,
+        };
+    }
+
+    /**
+     * Processes values from getScriptTagAttributes() to create HTML head element
+     * script tags with content, src (source), and data MIME type attributes.
+     *
+     * @method getHeadTags
+     * @param {ScriptTagsData} scriptTagsData Default values for script tags
+     * @return {HeadTagDef[]}
+     */
+    getHeadTags(scriptTagsData: ScriptTagsData): HeadTagDef[] {
+        const scriptTagDefs: ScriptTagDef | ScriptTagDef[] = this.getScriptTagAttributes(scriptTagsData);
+        const { src, type, dataContent } = scriptTagDefs;
+        const attrs: ScriptTagAttrs = this.makeScriptTagAttrs(src, type);
+        const array: HeadTagDef[] = [];
+        const tagType: string = TagType.SCRIPT as string;
+	    array.push({type: tagType, content: dataContent, attrs});
+        return array;
+    }
```

Comment:
All of these logic seems to be directly ported from the `meta-tags` service. The `meta-tags` service need `makeMetaTagsAttrs()`, `getMetaTags()` and `getHeadTags()` because it needs to do some processing of the data. More specifically, it needs to generate a lot of different meta tags for different consumers (FB, twitter, etc.) based on the same set of data.

Here, the use case is different: we only need to generate *one* <script> tag. No data transformation or processing is necessary. In fact, the processing done here is redundant, and here is why:

`getHeadTags()` takes in an object that looks like `{ type: 'abc', src: 'cde', content: 'efg' }` as a parameter. It then passes that parameter to `getScriptTagsAttributes()` method that returns an object that looks like  `{ type: 'abc', src: 'cde', dataContent: 'efg' }`. It then take out the `src` and `type` from the return object and pass them to `makeScriptTagAttrs()`, which returns `{ src: 'cde', type: 'abc' }`.  However, the `type` is never used, since what we are returning in this method is a one-element array that is `[ {type: TagType.SCRIPT, content: 'efg', src: 'cde' } ]`. If this is the case, it is not necessary to construct a service to do these in such roundabout fashion. Simply constructing an array that looks like `[{ type: TagType.SCRIPT, src: 'cde', content: 'efg' }]` would be sufficient.

Let's get rid of this service.

Reply:
Go ahead and read the previous comment and let me know if you still have questions. We do have a type attribute for this context, which is the application/ld+json value. Not having that in the script tag is problematic. Additionally, the src attribute is important for script tags. They have accessibility applications as well as providing an alternative should the JSON object not be retrievable.

Reply:
When you invoke `this.scriptTags.getHeadTags(scriptTagData)`, the parameter `scriptTagData` is something like this:
```
{
    type: 'application/ld+json',
    src: `/${parentId}/metadata/?format=google-dataset-json-ld`,
    content: 'some json string here',
};
```
And the return value of the `getHeadTags()` function is an array that looks like
```
[{
    type: 'script',
    attrs: {
        type: 'application/ld+json',
        src: `/${parentId}/metadata/?format=google-dataset-json-ld`,'
    }
    content: 'some json string here',
}};
```
You are simply constructing an array with one element from an object that carries basically the same information. I don't think you need to use two nested function calls with 40+ lines added to a service to accomplish that.

Reply:
Yes, you are reading the code correctly. However the information is not the same and the current function returns the data and constructs the tag in a way that is not appropriate for this application. There may be a way to decrease the overall lines. I will take a look and we can also discuss it during a meeting.